src/pathfinding_igvc/test/path-finding-test.cpp

```diff
@@ -0,0 +1,219 @@
+
+    nav_msgs::Path path_msg;
+    std::vector<geometry_msgs::PoseStamped> poses;
+    for (int i = 1; i < 10; i++) {
+        geometry_msgs::PoseStamped pose_stamped;
+        pose_stamped.pose.position.x = i;
+        pose_stamped.pose.position.y = i;
+        poses.push_back(pose_stamped);
+    }
+    path_msg.poses = poses;
+
+    geometry_msgs::Twist twist_msg =
+    PathFinding::pathToTwist(path_msg, xPos, yPos, orientation);
+    std::cout << "Far Start Pos" << std::endl;
+    std::cout << twist_msg.linear.x << std::endl;
+    std::cout << twist_msg.angular.z << std::endl;
+    std::cout << std::endl;
+    EXPECT_NEAR(0.63, twist_msg.linear.x, 0.01);
+    EXPECT_NEAR(2.36, twist_msg.angular.z, 0.01);
+}
+
+// Case where path consists of 2 sharp turns, one 90 deg left and one 90 deg
+// right later
+TEST(PathFinding, testSharpTurns) {
+    double xPos        = 0;
+    double yPos        = 0;
+    double orientation = 0;
+
+    nav_msgs::Path path_msg;
+    std::vector<geometry_msgs::PoseStamped> poses;
+    for (int i = 1; i < 10; i++) {
+        geometry_msgs::PoseStamped pose_stamped;
+        if (i <= 5) {
+            pose_stamped.pose.position.x = 0;
+            pose_stamped.pose.position.y = i;
+        } else {
+            pose_stamped.pose.position.x = i - 5;
+            pose_stamped.pose.position.y = 5;
+        }
+        poses.push_back(pose_stamped);
+    }
+    path_msg.poses = poses;
+
+    geometry_msgs::Twist twist_msg =
+    PathFinding::pathToTwist(path_msg, xPos, yPos, orientation);
+    std::cout << "Sharp turns" << std::endl;
+    std::cout << twist_msg.linear.x << std::endl;
+    std::cout << twist_msg.angular.z << std::endl;
+    std::cout << std::endl;
+    EXPECT_NEAR(0.763, twist_msg.linear.x, 0.01);
+    EXPECT_NEAR(1.49, twist_msg.angular.z, 0.01);
+}
+
+// Case where path consists of erratic direction and magnitude changes
+TEST(PathFinding, testErraticPath) {
+    double xPos        = 0;
+    double yPos        = 0;
+    double orientation = 0;
+
+    nav_msgs::Path path_msg;
+    std::vector<geometry_msgs::PoseStamped> poses;
+    for (int i = 1; i < 10; i++) {
+        geometry_msgs::PoseStamped pose_stamped;
+        if (i % 2 == 0) {
+            pose_stamped.pose.position.x = 2 * i;
+            pose_stamped.pose.position.y = -2 * i;
+        } else {
+            pose_stamped.pose.position.x = i;
+            pose_stamped.pose.position.y = 2 * i;
+        }
+        poses.push_back(pose_stamped);
+    }
+    path_msg.poses = poses;
+
+    geometry_msgs::Twist twist_msg =
+    PathFinding::pathToTwist(path_msg, xPos, yPos, orientation);
+    std::cout << "Erratic Path" << std::endl;
+    std::cout << twist_msg.linear.x << std::endl;
+    std::cout << twist_msg.angular.z << std::endl;
+    std::cout << std::endl;
+    EXPECT_NEAR(0.96, twist_msg.linear.x, 0.01);
+    EXPECT_NEAR(0.25, twist_msg.angular.z, 0.01);
+}
+
+// Case where position/orientation of robot is ahead of the path
```

Comment:
I really like the descriptive comments in front of each test, makes it much easier to quickly read through them all and see what sorts of things we're testing.